### PR TITLE
Qr Processing and Display

### PR DESCRIPTION
**Pull request by CraigP17**

### CODE REVIEW

File: client/src/components/QRData/QRData.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import axios from 'axios';
+import React, { useState, useEffect } from 'react';
+
+function QRData(props) {
+
+    const [qrData, setQrData] = useState({"success": false, "qr_type": "", "qr_data": {"is_found": false}});
+
+    useEffect(() => {
+        console.log(props.qrReady)
```
Comment (JacobRajah) [inline client/src/components/QRData/QRData.js]: Do we want to keep console logs?

File: client/src/components/QRData/QRData.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        axios.get(`/get_parsed_qr/${props.qrType}`)
+        .then((response) => {
+            console.log(response.data);
```
Comment (JacobRajah) [inline client/src/components/QRData/QRData.js]: Same comment

File: client/src/components/Task1/Task1.js [hunk 3]
```diff
@@ -34,7 +38,13 @@ function Task1() {
                     <Item>Component Here</Item>
                 </Grid>
                 <Grid xs={6}>
-                    <Item>Component Here</Item>
+                    <Item>
+                        <Grid container>
+                            {/* Display Processed QR Data */}
+                            <Grid item xs={6}><QRData qrType="1" qrReady={qrReady === "1"}/></Grid>
```
Comment (JacobRajah) [inline client/src/components/Task1/Task1.js]: I think qrReady is an int so should be qrReady === 1

File: server/app.py [hunk 4]
```diff
@@ -33,31 +44,47 @@ def task_2():
...
+    else:
+        return {
+            "success": False,
+            "message": "Invalid QR Type"
+        }
+
+
+@app.route('/get-telemetry', methods=['GET'])
+def get_telemetry():
     response = jsonify(data=telem[-1],
```
Comment (JacobRajah) [inline server/app.py]: Need to fix at some point, shouldn't be setting headers

File: server/app.py [hunk 5]
```diff
@@ -33,31 +44,47 @@ def task_2():
...
+
+    # Access POST request parameters
+    raw_qr_str = json_r["raw_qr_string"] if "raw_qr_string" in json_r else None
+    qr_type = int(json_r["qr_type"]) if "qr_type" in json_r else None
+
+    # QR Processing
+    if raw_qr_str and qr_type:
+        if QR_LST.qrs[qr_type - 1].is_valid(raw_qr_str):
+            return QR_LST.qrs[qr_type - 1].process(raw_qr_str)
+        return {"success": False, "message": f"Invalid QR {qr_type} Format"}
```
Comment (JacobRajah) [inline server/app.py:66]: Whenever I submit a QR, even if valid, this return is hit

File: server/app.py [hunk 6]
```diff
@@ -1,21 +1,32 @@
 from flask import Flask, jsonify, request
 
+from qr import AllQr
+
+app = Flask(__name__)
+app.config['CORS-HEADERS']: 'Content-Type'
```
Comment (JacobRajah) [inline server/app.py:8]: Can we remove this and switch to axis use? Should fix cors issue. Or is a new branch planned to fix this?

File: server/app.py (hunk 4 above)
Comment (CraigP17) [reply server/app.py]: I will create an Issue for a later fix

File: server/app.py (hunk 6 above)
Comment (CraigP17) [reply server/app.py:8]: Will be new issue/your event task

File: client/src/components/QRData/QRData.js (hunk 1 above)
Comment (CraigP17) [reply client/src/components/QRData/QRData.js]: Removed

File: client/src/components/QRData/QRData.js (hunk 2 above)
Comment (CraigP17) [reply client/src/components/QRData/QRData.js]: Removed

File: client/src/components/Task1/Task1.js (hunk 3 above)
Comment (CraigP17) [reply client/src/components/Task1/Task1.js]: Removed equality check

**Review by JacobRajah — APPROVED** [review]

**Pull request merged**